weblate/api/views.py

```diff
@@ -454,6 +454,8 @@ def file(self, request, **kwargs):
             request,
             serializer.validated_data['file'],
             serializer.validated_data['overwrite'],
+            serializer.validated_data['email'],
+            serializer.validated_data['author'],
```

Comment:
What it should additionally do here is to check permissions - not all users are allowed to set email and author, see related code in the UI:

https://github.com/WeblateOrg/weblate/blob/8b43331590c33ec19a14e6f214e103db94f8dfb3/weblate/trans/forms.py#L634

Reply:
Thank you for an update. I will try to implement also.